lib/dry/monads/registry.rb

```diff
@@ -0,0 +1,64 @@
+require 'concurrent/map'
+
+module Dry
+  # Common, idiomatic monads for Ruby
+  #
+  # @api private
+  module Monads
+    @registry = {}
+    @paths = {
+      do: 'dry/monads/do/all',
+      lazy: 'dry/monads/lazy',
+      list: 'dry/monads/list',
+      maybe: 'dry/monads/maybe',
+      task: 'dry/monads/task',
+      try: 'dry/monads/try',
+      validated: 'dry/monads/validated',
+      result: [
+        'dry/monads/result',
+        'dry/monads/result/fixed'
+      ]
+    }.freeze
+    @mixins = Concurrent::Map.new
```

Comment:
I used `Concurrent::Map` exclusively because of its API, `fetch_or_store` specifically. It's not that I expect any concurrency issues, IMO they need to be handled on a different layer (dry-system as an example).

Reply:
@flash-gordon makes sense, then!